internal/pkg/skuba/upgrade/node/versions.go

```diff
@@ -67,7 +68,11 @@ func (nviu NodeVersionInfoUpdate) IsUpdated() bool {
 
 func (nviu NodeVersionInfoUpdate) IsFirstControlPlaneNodeToBeUpgraded() bool {
 	isControlPlane := nviu.Current.IsControlPlane()
-	currentClusterVersion, _ := kubeadm.GetCurrentClusterVersion()
+	currentClusterVersion, err := kubeadm.GetCurrentClusterVersion()
+	if err != nil {
+		klog.Error(err.Error())
+		return false
```

Comment:
I'd change the signature of this function to `(bool, error)` and bubble up the error here... I think with this current behavior we could hit weird logic errors. WDYT?

Reply:
it would likely give false logic results in https://github.com/SUSE/skuba/blob/master/pkg/skuba/actions/node/upgrade/apply.go

but I would rather change the function like this:

```go
func (nviu NodeVersionInfoUpdate) IsFirstControlPlaneNodeToBeUpgraded(currentClusterVersion *version.Version) bool {
	isControlPlane := nviu.Current.IsControlPlane()
	allControlPlanesMatchVersion, _ := kubernetes.AllControlPlanesMatchVersion(currentClusterVersion)
	matchesClusterVersion := (currentClusterVersion.String() == nviu.Current.KubeletVersion.String())

	return isControlPlane && allControlPlanesMatchVersion && matchesClusterVersion
}
```

and pass the `currentClusterVersion` which is [already instantiated during the upgrade apply](https://github.com/SUSE/skuba/blob/master/pkg/skuba/actions/node/upgrade/apply.go#L50)

Reply:
Also, looking at that snippet, isn't `matchesClusterVersion := (currentClusterVersion.String() == nviu` redudant? I think `AllControlPlanesMatchVersion` already checks that, right?

So this could be refactored as `return isControlPlane && allControlPlanesMatchVersion` if I'm not mistaken.

Reply:
Done!